yarn-project/foundation/src/fields/point.ts

```diff
@@ -37,8 +37,7 @@ export class Point {
    * @returns A randomly generated Point instance.
    */
   static random() {
-    // TODO make this return an actual point on curve.
-    return new Point(Fr.random(), Fr.random(), false);
+    return Point.fromXAndSign(Fr.random(), randomBoolean());
```

Comment:
Sneaked this fix in as the new `fromXAndSign` func made it straightforward to do.